### PR TITLE
Make all Example project compile and work again

### PR DESCRIPTION
**Pull request by garnett**
There were plenty of reasons why each project did not work: 
* Deprecated APIs
* Xcode7 warnings
* etc.

I've updated all of them and now they work.

### CODE REVIEW

File: Examples/iPhone Custom UI Demo/Classes/iRateAppDelegate.m [hunk 1]
```diff
@@ -77,7 +77,8 @@ - (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)
     }
     else if (buttonIndex == 3) // open web page
     {
-        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"http://www.apple.com"]];
+        NSURL *url = [NSURL URLWithString:@"http://www.apple.com"];
```
Comment (nicklockwood) [inline Examples/iPhone Custom UI Demo/Classes/iRateAppDelegate.m]: Hard to see why this would prevent it from building o_O

File: iRate/iRate.m [hunk 2]
```diff
@@ -693,7 +693,8 @@ - (void)checkForConnectivityInBackground
 
         NSError *error = nil;
         NSURLResponse *response = nil;
-        NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:iTunesServiceURL] cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:REQUEST_TIMEOUT];
+        NSURL *url = [NSURL URLWithString:iTunesServiceURL];
```
Comment (nicklockwood) [inline iRate/iRate.m]: Or this...

File: Examples/iPhone Custom UI Demo/Classes/iRateAppDelegate.m (hunk 1 above)
Comment (garnett) [inline Examples/iPhone Custom UI Demo/Classes/iRateAppDelegate.m]: It is related to warning "using nullable type in nonnull context" thing

File: iRate/iRate.m (hunk 2 above)
Comment (garnett) [inline iRate/iRate.m]: Same here :)

**Pull request merged**